sktime/forecasting/statsmodels_arima.py

```diff
@@ -0,0 +1,316 @@
+        product of gradient estimator, 'oim' for the observed information matrix
+        estimator, calculated using the method of Harvey (1989), 'approx' for the
+        observed information matrix estimator, calculated using a numerical
+        approximation of the Hessian matrix, 'robust' for an approximate
+        (quasi-maximum likelihood) covariance matrix that may be valid even in the
+         presence of some misspecifications. Intermediate calculations use the 'oim'
+         method. 'robust_approx' is the same as 'robust' except that the intermediate
+         calculations use the 'approx' method. 'none' for no covariance matrix
+         calculation. Default is 'opg' unless memory conservation is used to avoid
+         computing the loglikelihood values for each observation,
+         in which case the default is 'oim'.
+
+    cov_kwds : dict or None, optional
+        A dictionary of arguments affecting covariance matrix computation.
+
+    return_params : bool, optional
+        Whether to return only the array of maximizing parameters. Default is False.
+
+    low_memory : bool, optional If set to True, techniques are applied to
+        substantially reduce memory usage. If used, some features of the results object
+        will not be available (including smoothed results and in-sample prediction),
+        although out-of-sample forecasting is possible. Default is False.
+
+    See Also
+    --------
+    ARIMA
+    AutoARIMA
+    StatsForecastAutoARIMA
+
+    Examples
+    --------
+    >>> from sktime.datasets import load_airline
+    >>> from sktime.forecasting.statsmodels_arima import StatsModelsARIMA
+    >>> y = load_airline()
+    >>> forecaster = StatsModelsARIMA(order=(0, 0, 12))  # doctest: +SKIP
+    >>> forecaster.fit(y)  # doctest: +SKIP
+    >>> y_pred = forecaster.predict(fh=[1,2,3])  # doctest: +SKIP
+    """
+
+    _tags = {
+        # packaging info
+        # --------------
+        "authors": ["arnaujc91"],
+        "maintainers": ["arnaujc91"],
+        "ignores-exogeneous-X": False,
+        "capability:pred_int": True,
+        "capability:pred_int:insample": True,
+        "python_dependencies": ["statsmodels"],
+    }
+
+    def __init__(
+        self,
+        order: Tuple[int, int, int] = (0, 0, 0),
+        seasonal_order: Tuple[int, int, int, int] = (0, 0, 0, 0),
+        trend: Optional[Union[str, Iterable]] = None,
+        enforce_stationarity: bool = True,
+        enforce_invertibility: bool = True,
+        concentrate_scale: bool = False,
+        trend_offset: int = 1,
+        dates: Optional[np.ndarray] = None,
+        freq: Optional[str] = None,
+        missing: Optional[str] = None,
+        validate_specification: bool = True,
+        start_params: Optional[np.ndarray] = None,
+        transformed: bool = True,
+        includes_fixed: bool = False,
+        method: Optional[str] = None,
+        method_kwargs: Optional[dict] = None,
+        gls: Optional[bool] = False,
```

Comment:
For consistent type hint with `enforce_stationarity`, `enforce_invertibility`, `concentrate_scale` etc. above, I think we should not hint this as an optional field.